File: JDBCReview/src/main/java/com/atguigu/jdbcutils/JDBCUtils.java
package com.atguigu.jdbcutils;

import com.alibaba.druid.pool.DruidDataSourceFactory;
import com.atguigu.jdbc.DbUtilsTest;
import org.junit.Test;

import javax.sql.DataSource;
import java.io.InputStream;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.Properties;

/**
 * @author Yu HaiFeng
 * @Description
 * @create 2020-05-22 7:18
 */
public class JDBCUtils {

    private static DataSource dataSource;

    static{
        try {
            InputStream resourceAsStream = DbUtilsTest.class.getClassLoader().getResourceAsStream("druid.properties");
            Properties properties = new Properties();
            properties.load(resourceAsStream);

            dataSource = DruidDataSourceFactory.createDataSource(properties);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }


    public static  Connection getConnection(){
        Connection connection = null;
        try {
            connection = dataSource.getConnection();
        } catch (SQLException e) {
            e.printStackTrace();
        }

        return connection;
    }

    public static void closeResource(Connection connection){
        if (connection != null){
            try {
                connection.close();
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }


    }
}

File: Summer/src/main/scala/com/atguigu/summer/utils/JDBCUtils.scala
package com.atguigu.summer.utils

import java.sql.Connection
import java.util

import com.alibaba.druid.pool.DruidDataSourceFactory
import javax.sql.DataSource


/**
 * @Description
 * @author Yu HaiFeng
 * @create 2020-06-17 0:57
 */
object JDBCUtils {

    
    var dataSource: DataSource = init()
    
    def init(): DataSource = {
        val paramMap = new util.HashMap[String,String]()
        paramMap.put("driverClassName",PropertiesUtil.getValue("jdbc.driver.name"))
        paramMap.put("url",PropertiesUtil.getValue("jdbc.url"))
        paramMap.put("username",PropertiesUtil.getValue("jdbc.username"))
        paramMap.put("password",PropertiesUtil.getValue("jdbc.password"))
        paramMap.put("maxActive",PropertiesUtil.getValue("jdbc.datasource.size"))
    
    
        val source: DataSource = DruidDataSourceFactory.createDataSource(paramMap)
        source
    }
    
    def getConnection: Connection = {
        dataSource.getConnection()
    }
}
